test/libethereum/blockchainHelper.cpp

```diff
@@ -0,0 +1,239 @@
+			m_state.sync(*m_blockchain);
+			m_state.sync(*m_blockchain, txs, gp);
+			mine(m_state, *m_blockchain);
+		}
+		catch (Exception const& _e)
+		{
+			cnote << "state sync or mining did throw an exception: " << diagnostic_information(_e);
+			return;
+		}
+		catch (std::exception const& _e)
+		{
+			cnote << "state sync or mining did throw an exception: " << _e.what();
+			return;
+		}
+
+		//make actual block from it
+		Transactions txList;
+		for (auto const& txi: txs.transactions()) //get valid transactions
+			txList.push_back(txi.second);
+
+		RLPStream txStream;
+		txStream.appendList(txList.size());
+		for (unsigned i = 0; i < txList.size(); ++i)
+		{
+			RLPStream txrlp;
+			txList[i].streamRLP(txrlp);
+			txStream.appendRaw(txrlp.out());
+		}
+
+		//Uncles
+		RLPStream uncleStream(0);
+		BlockQueue uncleBlockQueue;
+
+		//Make full block stream (with transactions and uncles)
+		RLPStream rlpStream;
+		m_state.info().streamRLP(rlpStream, WithNonce);
+
+		RLPStream blockStream(3);
+		blockStream.appendRaw(rlpStream.out());
+		blockStream.appendRaw(txStream.out());
+		blockStream.appendRaw(uncleStream.out());
+
+		//store block to the blockchain
+		try
+		{
+			m_blockchain->sync(uncleBlockQueue, m_state.db(), 4);
+			m_blockchain->attemptImport(blockStream.out(), m_state.db());
+			m_state.sync(*m_blockchain);
+		}
+		catch (Exception const& _e)
+		{
+			cnote << "block sync or import did throw an exception: " << diagnostic_information(_e);
+			m_testBlocks.pop_back();
+			return;
+		}
+		catch (std::exception const& _e)
+		{
+			cnote << "block sync or import did throw an exception: " << _e.what();
+			m_testBlocks.pop_back();
+			return;
+		}
+	}
+
+	void rebuild()
+	{
+		delete m_blockchain;
+		m_blockchain = new dev::eth::CanonBlockChain(m_td.path(), WithExisting::Verify);
+	}
+
+private:
+	dev::TransientDirectory m_td; //BlockChain Dir
+	dev::TransientDirectory m_td2; //State Dir
+	std::string m_genesisRLPString = "0xf901fcf901f7a00000000000000000000000000000000000000000000000000000000000000000a01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347948888f1f195afa192cfee860698584c030f4c9db1a07dba07d6b448a186e9612e5f737d1c909dce473e53199901a302c00646d523c1a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421b90100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008302000080832fefd8808454c98c8142a0b28caac6b4c3c1f231ff6e4210a78d3b784988be77d89095038b843fb04440408821dbba66d491fb5ac0c0";
```

Comment:
Looks like const and static to me. Can you also split the literal into multiple lines?